### ISSUE TITLE
generate thumbnail only for first page

### ISSUE THREAD

**Issue by freestyle68** [codelibs/fess#1166]
Hi again,

I noticed that the generate-thumbnail script make thumbnail of every page of the pdf/office documents.
You can see it in /var/lib/fess/thumbnails/

This is useless because only the first page is supposed to be showed on thumbnails results. And this kill the server, imagine a 300 page document... in fact gs kill the server with this task:

gs -sstdout=%stderr -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 -sDEVICE=pngalpha -dTextAlphaBits=4 ..........

As a solution, you should add the flag

-e PageRange=1-1

to the commandline

unoconv -o $TMP_FILE -f pdf $TARGET_FILE

More details on

http://bernaerts.dyndns.org/linux/76-gnome/325-gnome-shell-generate-msoffice-thumbnail-nautilus

Another suggested feature is to add thumbnail also for image files, you could use

mogrify for jpg, png and other formats:

https://superuser.com/questions/844912/generate-thumbnail-with-imagemagick

**Comment by marevol** [comment]
Thank you for the info.
I'll add PageRange.

**Comment by freestyle68** [comment]
Hi,

even after fix  https://github.com/codelibs/fess/issues/1168 still PNGs for every document page are generated in /var/lib/fess/thumbnails/

I tried the script

`/usr/share/fess/bin/generate-thumbnail msoffice <MS Office file> <Output Path>
`

for office and pdf files and this correctly generate a single pdf or png of the first page.

But then the process start:

gs -sstdout=%stderr -dQUIET ....  -sOutputfile/tmp-magik-.......

and is this that generate the mess. This also is a 100% CPU process that lasts a lot of time.

Also, even after this procedure the thumbnails aren't visible on search results.
If I use the chrome developer tool to analize the missing thumbnail, the result is the following:

```
<div class="thumbnailBox media-left hidden-xs-down">
<a class="link" href="file://dati/few/064/064452.pdf" data-uri="file://dati/few/064/064452.pdf" data-id="22d2c3a975f049728042aacc6bfe9bb7" data-order="0">
<img src="/images/noimage.png" data-src="/thumbnail/?docId=22d2c3a975f049728042aacc6bfe9bb7&amp;queryId=8919116e193244ae8120a7cd365784fb" class="thumbnail" style="background-image: url(&quot;/images/loading.gif&quot;);">
```

so the img src is not pointing to image in /var/lib/fess/thumbnails/

**Comment by marevol** [comment]
What is the server hardware spec?

**Comment by freestyle68** [comment]
Xeon with 16 GB RAM. Debian 9.
Also tried with Centos 7, same results.

The gs process put a single thread to 100% CPU, not all threads.

Crawled now 27 files (docs and pdfs) and after 15 min still active Thumbnail generator job. No thumbnail visible on search results.

Same errors in /var/log/fess

Tried also with a new from scratch VPS, Debian 9, OpenJDK, 16 GB, but same problem.

**Comment by marevol** [comment]
Thank you for the info.
I think it's improved by #1173.

**Comment by freestyle68** [comment]
YES, now the generate-thumbnail task is very fast and the thumbnails are visible on the search results.

But only for PDF files.
For office files thumbnails aren't visible.

This is not dued to missing thumbnail but to a web config problem: in fact with office docs I cannot see the blank image (img src="/images/noimage.png) usually visible before generate thumbnail job.
This image instead is visible with PDF files before thumbnail job.

**Comment by freestyle68** [comment]
Perfect! The fix https://github.com/codelibs/fess/issues/1175 solved the problem.

I tried crawling about 6000 docs and I have no problem.
But I have changed the timeout from 10 to 30 sec on

`src/main/java/org/codelibs/fess/thumbnail/impl/CommandGenerator.java
`

because a lot of

`[CommandGeneratorDestoryTimer-1500312534655] WARN CommandGenerator is timed out
`

errors (slow disk). This caused the thumbnail job to abort and restart a few minutes later.
Thank you very much.
